features/bsm/service/impl/src/main/java/org/opennms/netmgt/bsm/service/model/graph/internal/BusinessServiceGraphImpl.java

```diff
@@ -135,15 +136,25 @@ public Void visit(ApplicationEdge edge) {
                         addVertex(vertexForEdge[0]);
                         m_verticesByApplicationId.put(vertexForEdge[0].getApplication().getId(), vertexForEdge[0]);
 
-                        for (String reductionKey : edge.getReductionKeys()) {
-                            GraphVertex reductionKeyVertex = m_verticesByReductionKey.get(reductionKey);
-                            if (reductionKeyVertex == null) {
-                                reductionKeyVertex = new GraphVertexImpl(REDUCE_HIGHEST_SEVERITY, reductionKey);
-                                addVertex(reductionKeyVertex);
-                                m_verticesByReductionKey.put(reductionKey, reductionKeyVertex);
+                        for (IpService ipService : edge.getApplication().getIpServices()) {
```

Comment:
This comment was not addressed. 
Just checking, if it was overseen, or is there just no better way of implementing this?

Reply:
This comment was addressed. The duplicated code was moved into a separate method. However, since the "for" statement of line 127 was not modified this comment looks like it was not addressed but of course it is.